src/components/Toolbar/ToolbarComponent.tsx

```diff
@@ -1,16 +1,24 @@
 import React from "react";
 import { useAlgorithmRunner } from "../../hooks/useAlghorithmRunner";
-import { RunButton, RunCodeIcon, StyledToolbar } from "./styles";
+import { useAlgorithmRunnerWithSteps } from "../../hooks/useAlgorithmRunnerWithSteps";
+import { RunButton, RunCodeIcon, NextButton, NextStepIcon, StyledToolbar } from "./styles";
 import { ToolbarProps } from "./ToolbarProps";
 
 function ToolbarF(props: ToolbarProps, ref?: React.Ref<HTMLDivElement>): JSX.Element {
   const [repeatLastRun] = useAlgorithmRunner();
+  const generator = useAlgorithmRunnerWithSteps();
+
   return (
     <StyledToolbar {...props} ref={ref}>
       <RunButton onClick={repeatLastRun}>
         <RunCodeIcon />
         <span>Repeat last run</span>
       </RunButton>
+
+      <NextButton onClick={() => generator.next()}>
+        <NextStepIcon />
+        <span>Next step</span>
+      </NextButton>
```

Comment:
Next step button odpala ten runWithRestarts, jednak nie wiem, czy nie byłoby słuszne, żeby przycisk miał na początku napis "Run with steps" ("aktywowanie" przycisku, udające tworzenie tego generatora (bo generator już jest stworzony)), a po kliknieciu zmienić na "Next step". Jak myślicie?

Reply:
Było by słuszne. 
Ale myślę, że nie w tym PR.
Myślę, że będzie osobne zadanie na jakąś integrację tego z konsolą i tam można będzie też dodać to o czym ty mówisz ;) (chyba że chcesz dodać tę obsługę)